libs/utils/utils.js

```diff
@@ -500,50 +507,54 @@ export function decorateAutoBlock(a) {
     window.lana?.log(`Cannot make URL from decorateAutoBlock - ${a?.href}: ${e.toString()}`);
     return false;
   }
-  const href = hostname === url.hostname ? `${url.pathname}${url.search}${url.hash}` : a.href;
+
+  const href = hostname === url.hostname
+    ? `${url.pathname}${url.search}${url.hash}`
+    : a.href;
+
   return config.autoBlocks.find((candidate) => {
     const key = Object.keys(candidate)[0];
     const match = href.includes(candidate[key]);
-    if (match) {
-      if (key === 'pdf-viewer' && !a.textContent.includes('.pdf')) {
-        a.target = '_blank';
-        return false;
-      }
+    if (!match) return false;
 
-      if (key === 'fragment' && url.hash === '') {
-        const { parentElement } = a;
-        const { nodeName, innerHTML } = parentElement;
-        const noText = innerHTML === a.outerHTML;
-        if (noText && nodeName === 'P') {
-          const div = createTag('div', null, a);
-          parentElement.parentElement.replaceChild(div, parentElement);
-        }
-      }
-
-      // previewing a fragment page with mp4 video
-      if (key === 'fragment' && a.textContent.match('media_.*.mp4')) {
-        a.className = 'video link-block';
-        return false;
-      }
+    if (key === 'pdf-viewer' && !a.textContent.includes('.pdf')) {
+      a.target = '_blank';
+      return false;
+    }
 
-      // Modals
-      if (key === 'fragment' && url.hash !== '') {
-        a.dataset.modalPath = url.pathname;
-        a.dataset.modalHash = url.hash;
-        a.href = url.hash;
-        a.className = 'modal link-block';
-        return true;
+    const isInlineFrag = url.hash.includes('#_inline');
```

Comment:
The if block on line 526 needs to be run before the `media_.*mp4` check since the fragment could contain an mp4 in the path.  However I think it may be better to wrap the 3 `if` checks inside a `if (key === 'fragment')` block.  I'll give that a try.

Honestly I don't know if that is a valid use case, but I don't want to change the execution order.

Reply:
fair